heartbeat/ocf-distro

```diff
@@ -5,43 +5,205 @@
 # released RH specific stuff (/etc/sysconfig/nfs editing)
 # 
 
+# systemd-based systems should all have an os-release file.
+_ETC_OS_RELEASE_FILE="/etc/os-release"
+_USR_OS_RELEASE_FILE="/usr/lib/os-release"
+
+# Legacy distro-specific files
 _DEBIAN_VERSION_FILE="/etc/debian_version"
 _REDHAT_RELEASE_FILE="/etc/redhat-release"
 _SUSE_RELEASE_FILE="/etc/SuSE-release"
-_RELEASE_FILES="/etc/*-release"
-_REDHAT_BASED_DISTROS_RE='red *hat|fedora|centos|scientific linux'
 
-get_release_id() {
+_DEBIAN_BASED_DISTROS_RE="debian|ubuntu"
+_REDHAT_BASED_DISTROS_RE="red *hat|rhel|fedora|centos|scientific"
+_SUSE_BASED_DISTROS_RE="sles|suse"
+
+
+# Convert OS release ID to a standard form regardless of source.
+_process_os_release_id() {
+	_os="$1"
+
+	# Isolate distro name, remove whitespace, convert to lowercase
+	_os=$(echo "$_os" | sed -e "s/\([G][Nn][Uu]\/\)\?[Ll][Ii][Nn][Uu][Xx]//" \
+		-e "s/[Ss][Ee][Rr][Vv][Ee][Rr]//" \
+		-e "s/release.*//" -e "s/[[:digit:]].*//" \
+		| tr -d '[:blank:]' | tr '[:upper:]' '[:lower:]')
```

Comment:
You can also do these 2 as part of the final sed statement (e.g. using `sed -e "s/\([^ [:digit:]]\+\).*/\L\1/i"` or similar).

Reply:
Not POSIX-compliant. But like I said in another response, neither is `\?`, so I'll need to remove that and split `GNU/Linux` into two separate expressions.

I made an effort to keep everything portable, but it wasn't until tonight that I noticed the `--posix` flag for `sed` to actually verify these things.

Reply:
I must not have been thinking straight when I did this in the first place. All I needed to do was move the lowercase conversion to the beginning.

This is simplified in the latest push.